framework/core/src/Discussion/Utf8SlugDriver.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Discussion;
+
+use Flarum\Database\AbstractModel;
+use Flarum\Http\SlugDriverInterface;
+use Flarum\User\User;
+
+class Utf8SlugDriver implements SlugDriverInterface
+{
+    /**
+     * @var DiscussionRepository
+     */
+    protected $discussions;
+
+    public function __construct(DiscussionRepository $discussions)
+    {
+        $this->discussions = $discussions;
+    }
+
+    public function toSlug(AbstractModel $instance): string
+    {
+        $title = preg_replace('![-\s]+!u', '-', $instance->title);
```

Comment:
we should probably escape non-word characters though I think

Reply:
Would \W and \D be better so that anything that's not a Digit or Word Character is replaced? From my knowledge these properly support UTF-8?

Reply:
I think this is the same as https://github.com/flarum/framework/blob/main/framework/core/src/Discussion/Search/Gambit/FulltextGambit.php#L28, since we wouldn't want to be removing the characters we're trying to support.